### PR TITLE
SHIELD-185 HA show certs command to get certificates for all services

### PR DESCRIPTION
**Pull request by atultherajput**
Signed-off-by: Atul Krishna <Atul.Krishna@progress.com>

### :nut_and_bolt: Description: What code changed, and why?
Added cert show command.

### :chains: Related Resources
https://chefio.atlassian.net/browse/SHIELD-185
### :white_check_mark: Checklist

**All PRs** must tick these:

- [X] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [X] All commits signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

With occasional exceptions, all PRs **from Progress employees** must tick these:

- [X] Is the code clear? *(complicated code or lots of comments--subdivide and use well-named methods, meaningful variable names, etc.)*
- [X] Consistency checked? *(user notifications, user prompts, visual patterns, code patterns, variable names)*
- [X] Repeated code blocks eliminated? *(adapt and reuse existing components, blocks, functions, etc.)*
- [X] Spelling, grammar, typos checked? *(at a minimum use `make spell` in any component directory)*
- [X] Code well-formatted? *(indents, line breaks, etc. improve rather than hinder readability)*

All PRs **from Progress employees** should tick these if appropriate:

- [X] Tests added/updated? (all new code needs new tests)
- [X] Docs added/updated? (all customer-facing changes)

*Please add a note next to any checkbox above if you are NOT ticking it.*

### :camera: Screenshots, if applicable

Screenshot 2022-12-09 at 4 46 54 PM

---------------

Screenshot 2022-12-09 at 4 47 35 PM

---------------

Screenshot 2022-12-09 at 4 48 46 PM

---------------

Screenshot 2022-12-09 at 4 49 27 PM

---------------

Screenshot 2022-12-09 at 4 50 20 PM

### CODE REVIEW

File: components/automate-cli/cmd/chef-automate/ha_cert_show.go [hunk 1]
```diff
@@ -0,0 +1,414 @@
...
+
+	certShowCmd := &cobra.Command{
+		Use:   "show",
+		Short: "Chef Automate Certificates Show",
+		Long:  "Chef Automate CLI command to show certificates, this command should always be executed from AutomateHA Bastion Node",
+		RunE:  certShowCmdFunc(&flagsObj),
+	}
+
+	certShowAutomateCmd := &cobra.Command{
+		Use:   "automate",
```
Comment (Dmaddu) [inline components/automate-cli/cmd/chef-automate/ha_cert_show.go:66]: we are following some patterns to refer to the services as flags, so I recommend using the same flags.
https://deploy-preview-7604--chef-automate.netlify.app/automate/cli_chef_automate/#chef-automate-cert-rotate

File: components/automate-cli/cmd/chef-automate/ha_cert_show.go [hunk 2]
```diff
@@ -0,0 +1,414 @@
...
+		c.printOpensearchCertificates(certInfo)
+	}
+}
+
+// printAutomateCertificates prints automate certificates
+func (c *certShowImpl) printAutomateCertificates(certInfo certShowCertificates) {
+	c.writer.Title("Automate Certificates")
+	c.writer.HR()
+
+	c.writer.Println("========================Automate Root CA========================")
```
Comment (Dmaddu) [inline components/automate-cli/cmd/chef-automate/ha_cert_show.go:238]: Are we got the approval to show a response like this to the customers?

File: components/automate-cli/cmd/chef-automate/ha_cert_show.go [hunk 3]
```diff
@@ -0,0 +1,414 @@
...
+	c.writer.Println(fmt.Sprintf("\n======================%s Private Key======================", remoteService))
+	if len(strings.TrimSpace(certs.PrivateKey)) == 0 {
+		c.writer.Println(fmt.Sprintf("No %s private key found", remoteService))
+	} else {
+		c.writer.Println(certs.PrivateKey)
+	}
+}
+
+// validateNode validates if the node exists in the cluster
+func (c *certShowImpl) validateNode(certs []CertByIP, remoteService string) error {
```
Comment (Dmaddu) [inline components/automate-cli/cmd/chef-automate/ha_cert_show.go:382]: is this function to validate the given IP belongs to the cluster?
If so we already implemented that in cert-rotate, can we use that?

**Review by Dmaddu — CHANGES_REQUESTED** [review]

File: components/automate-cli/cmd/chef-automate/ha_cert_show.go [hunk 4]
```diff
@@ -0,0 +1,414 @@
...
+	return certShowImpl{
+		flags:     flags,
+		nodeUtils: nodeUtils,
+		sshUtil:   sshUtil,
+		writer:    writer,
+	}
+}
+
+// certShowCmdFunc is the main function for the cert show command
+func certShowCmdFunc(flagsObj *certShowFlags) func(cmd *cobra.Command, args []string) error {
```
[the remark this answers is not in the document]
Comment (atultherajput) [reply components/automate-cli/cmd/chef-automate/ha_cert_show.go:98]: done

File: components/automate-cli/cmd/chef-automate/ha_cert_show.go (hunk 2 above)
Comment (atultherajput) [reply components/automate-cli/cmd/chef-automate/ha_cert_show.go:195]: No, we haven't. Please let me know whom to contact for this.

File: components/automate-cli/cmd/chef-automate/ha_cert_show.go (hunk 3 above)
Comment (atultherajput) [reply components/automate-cli/cmd/chef-automate/ha_cert_show.go:339]: We cannot use it directly as implementation differs based on the signature. So we have to modify the original function in order to use it.

**Review by Dmaddu — CHANGES_REQUESTED** [review]

File: components/automate-cli/cmd/chef-automate/ha_cert_show.go [hunk 5]
```diff
@@ -0,0 +1,379 @@
...
+	c.writer.Println(fmt.Sprintf("\n======================%s Private Key======================", remoteService))
+	if len(strings.TrimSpace(certs.PrivateKey)) == 0 {
+		c.writer.Println(fmt.Sprintf("No %s private key found", remoteService))
+	} else {
+		c.writer.Println(certs.PrivateKey)
+	}
+}
+
+// validateNode validates if the node exists in the cluster
+func (c *certShowImpl) validateNode(certs []CertByIP, remoteService string) error {
```
[the remark this answers is not in the document]
Comment (atultherajput) [reply components/automate-cli/cmd/chef-automate/ha_cert_show.go:336]: done

File: components/automate-cli/cmd/chef-automate/ha_cert_show.go [hunk 6]
```diff
@@ -0,0 +1,379 @@
...
+	}
+
+	for _, certs := range certInfo.AutomateCertsByIP {
+		c.printPublicAndPrivateKeys(certs, "Automate", true)
+	}
+
+}
+
+// printChefServerCertificates prints the chef server certificates
+func (c *certShowImpl) printChefServerCertificates(certInfo certShowCertificates) {
```
[the remark this answers is not in the document]
Comment (atultherajput) [reply components/automate-cli/cmd/chef-automate/ha_cert_show.go:217]: They are not almost the same as Automate has root CA while Chef Server does not. Similarly, OpenSearch has Admin Key and Admin Cert while Postgresql does not. So to combine them we have to add a lot of conditional statements and that will increase Cognitive Complexity.

File: components/automate-cli/cmd/chef-automate/certRotate_test.go [hunk 7]
```diff
@@ -13,6 +13,9 @@ const (
 	RemoteFilePath = "198.51.100.0:/home/ec2-user/certs/public.pem"
 	LocalFilePath  = "/home/ec2-user/certs/public.pem"
 	ValidIP        = "198.51.100.0"
+	ValidIP1       = "198.51.100.1"
+	ValidIP2       = "198.51.100.2"
+	ValidIP3       = "198.51.100.3"
```
Comment (Dmaddu) [inline components/automate-cli/cmd/chef-automate/certRotate_test.go:18]: This is not used in the code, please remove it.

**Review by Dmaddu — CHANGES_REQUESTED** [review]

File: components/automate-cli/cmd/chef-automate/ha_cert_show.go (hunk 6 above)
[the remark this answers is not in the document]
Comment (Dmaddu) [reply components/automate-cli/cmd/chef-automate/ha_cert_show.go:217]: @atultherajput  lets have a call on this

File: components/automate-cli/cmd/chef-automate/certRotate_test.go (hunk 7 above)
Comment (atultherajput) [reply components/automate-cli/cmd/chef-automate/certRotate_test.go]: done

File: components/automate-cli/cmd/chef-automate/ha_cert_show.go (hunk 6 above)
[the remark this answers is not in the document]
Comment (atultherajput) [reply components/automate-cli/cmd/chef-automate/ha_cert_show.go]: done.

**Review by AnkurMundhra — APPROVED** [review]
I have reviewed the screenshots attached with the PR>

**Review by vivekshankar1 — APPROVED** [review]

**Pull request merged**